main_windows.go

```diff
@@ -59,3 +61,7 @@ func GetAzCopyAppPath() string {
 	}
 	return azcopyAppDataFolder
 }
+
+func init() {
+	http.DefaultTransport.(*http.Transport).Proxy = common.ProxyMiddleman()
```

Comment:
This is just a backup, or cases that actually use the DefaultTransport, right? In general AzCopy creates new Transports, instead of using the default one, and in those sites where it makes new transports, you've put a call to ProxyMiddleman() right?
If that's correct, can you pop a comment here, to explain that?  Since otherwise this line could confuse future maintainers (they may think it accomplishes more than it actually does).